core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -118,14 +118,16 @@ private static void physicalFactAndMeasureMetadataDeletion(CarbonTable carbonTab
       if (canDeleteThisLoad(oneLoad, isForceDelete, cleanStaleInProgress)) {
         try {
           if (oneLoad.getSegmentFile() != null) {
-            String tablePath = carbonTable.getAbsoluteTableIdentifier().getTablePath();
-            Segment segment = new Segment(oneLoad.getLoadName(), oneLoad.getSegmentFile());
-            // No need to delete physical data for external segments.
-            if (oneLoad.getPath() == null || oneLoad.getPath().equalsIgnoreCase("NA")) {
-              SegmentFileStore.deleteSegment(tablePath, segment, specs, updateStatusManager);
+            if (canSegmentLockBeAcquired(oneLoad, carbonTable.getAbsoluteTableIdentifier())) {
```

Comment:
move this method call in canDeleteThisLoad().. and refactor the other places as well

Reply:
done